pkgs/by-name/kl/klong/package.nix

```
@@ -0,0 +1,52 @@
+{
+  fetchFromSourcehut,
+  fetchurl,
+  installShellFiles,
+  lib,
+  makeWrapper,
+  stdenv,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "klong";
+  version = "20221212";
+
+  src = fetchurl {
+    url = "https://t3x.org/klong/klong${version}.tgz";
+    hash = "sha256-XhpIdyKKPGQ6mdv9LXPmC8P6hW4mFawv54yANw5/lrQ=";
+  };
+
+  docs = fetchFromSourcehut {
+    owner = "~nut";
+    repo = "klong-docs";
+    rev = "350da558709e3728df60ddf45fafe09e3fb89139";
+    hash = "sha256-yfvXljjJwCETWPa70zXhaQJOHhZYR2k+BKAd0Dw/U70=";
+  };
+
+  nativeBuildInputs = [
+    makeWrapper
+    installShellFiles
+  ];
+
+  installPhase = ''
+    runHook preInstall
+    install -m 555 -Dt $out/bin kg kplot
+    runHook postInstall
+  '';
+
+  postFixup = ''
+    installManPage $docs/*.1
+    mkdir -p $out/lib/klong
+    cp lib/*.kg $out/lib/klong
+    wrapProgram $out/bin/kg --prefix KLONGPATH : $out/lib/klong
+  '';
+
+  meta = {
+    description = "Simple Array programming language";
+    homepage = "https://t3x.org/klong";
+    license = lib.licenses.publicDomain;
```

```suggestion
    license = with lib.licenses; [ publicDomain cc0 ];
```
klong allows both public domain and CC0.

> Nobody owns this code.
> It's in the public domain or whatever you call it.
> 
> If your jurisdiction does not offer the concept of the
> public domain, the Creative Commons Zero (CC0) license applies.
> See the file CC0 for a copy of the CC0 license.